src/components/menu/menu-trigger/menu-trigger.directive.ts

```diff
@@ -119,7 +120,7 @@ export class MenuTriggerDirective implements OnInit, OnDestroy {
         // position then it will still take precendence
         if (this._isSubmenuTrigger) {
             this.menu._isSubMenu = true;
-            this.menu.placement = 'right';
+            this.menu.placement = this.subMenuPlacement(this.menu.placement);
         }
```

Comment:
Does moving this into `ngOnChanges` solve the problem without duplicating this code?